python/ray/data/tests/test_dataset.py

```diff
@@ -3251,6 +3326,29 @@ def test_groupby_simple_std(ray_start_regular_shared):
     assert ray.data.from_items([3]).std() == 0
 
 
+def test_groupby_simple_multilambda(ray_start_regular_shared):
+    # Test built-in mean aggregation
+    seed = int(time.time())
+    print(f"Seeding RNG for test_groupby_simple_multilambda with: {seed}")
+    random.seed(seed)
+    xs = list(range(100))
+    random.shuffle(xs)
+    agg_ds = ray.data.from_items([[x, 2*x] for x in xs]) \
```

Comment:
Hmm we should probably parameterize all groupby tests by partition size (e.g., 1, 10, 100 partitions) since partitioning affects correctness.